src/components/CohortBuilder/SearchById/SearchByIdModal.js

```diff
@@ -0,0 +1,224 @@
+  handleClose() {
+    this.setState({ loading: false });
+    this.props.closeModal();
+  }
+
+  handleClear() {
+    this.setInputText('');
+  }
+
+  renderHeader() {
+    return (
+      <React.Fragment>
+        <ModalTitle>Upload a List of Identifiers</ModalTitle>
+        <CloseIcon
+          css="cursor:pointer; width:22px; height:22px;"
+          fill="black"
+          onClick={this.handleClose}
+        />
+      </React.Fragment>
+    );
+  }
+
+  renderBody() {
+    const { inputIdsText, inputIds, results, loading } = this.state;
+
+    return (
+      <React.Fragment>
+        <section className="sbi-description">
+          <p>
+            Type or copy-and-paste a list of comma delimited identifiers (participant, biospecimen,
+            file)
+          </p>
+          <WhiteButton key="cancel" onClick={this.handleClear} className="clear">
+            Clear
+          </WhiteButton>
+        </section>
+        <section className="sbi-id-input">
+          <textarea
+            placeholder="e.g. PT_X25FNZ4D, CDH1363, BS_E28336C7, 4-28F, GF_9R86WD1Z"
+            rows="4"
+            value={inputIdsText}
+            onChange={this.handleInputIdsChange}
+          />
+        </section>
+        <section className="sbi-description">
+          <span>Or choose file to upload</span>
+          <input
+            type="file"
+            multiple
+            accept=".tsv,.csv,text/*"
+            style={{ display: 'none' }}
+            onChange={this.handleFilesUpload}
+            ref={this.fileInpuRef}
+          />
+          <TealActionButton
+            className="btn-upload"
+            onClick={() => {
+              this.fileInpuRef.current.click();
+            }}
+          >
+            Upload csv
+          </TealActionButton>
+        </section>
+        {results === null ? null : (
+          <section className="sbi-results">
+            <hr />
+            <H3>Matching Participants</H3>
+            <SearchResults query={inputIds} results={results.participants} loading={loading} />
+          </section>
+        )}
+      </React.Fragment>
+    );
+  }
+
+  renderFooter() {
+    const { loading, inputIds, results } = this.state;
+    return (
+      <React.Fragment>
+        <WhiteButton key="cancel" onClick={this.handleClose}>
+          Cancel
+        </WhiteButton>
+        <TealActionButton
+          key="confirm"
+          disabled={loading === true || inputIds.length === 0 || results !== null}
+          onClick={this.handleViewResultsClick}
+        >
+          View Results
+        </TealActionButton>
+        <TealActionButton
+          key="apply"
+          disabled={loading === true || inputIds.length === 0 || results === null}
+          onClick={this.handleApplyFilterClick}
+        >
+          Apply Filter
+        </TealActionButton>
+      </React.Fragment>
+    );
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <Row className="header">{this.renderHeader()}</Row>
```

Comment:
It s really a matter of personal choice but I think I would make a component for each render (renderHeader, renderFooter,...) methods